File: akka/src/main/scala/com/ws/akka/scala/demo01/Demo01.scala
package com.ws.akka.scala.demo01

import akka.actor.{Actor, ActorLogging, ActorSystem, Props}

import scala.io.StdIn

/**
  *
  * <p>
  * 类名称：Demo01
  * </p>
  * <p>
  * 类描述：${DESCRIPTION}
  * </p>
  * <p>
  * 创建人：sun
  * </p>
  * <p>
  * 创建时间：2018-08-29 18:03
  * </p>
  * <p>
  * 修改人：
  * </p>
  * <p>
  * 修改时间：
  * </p>
  * <p>
  * 修改备注：
  * </p>
  * <p>
  * Copyright (c) 版权所有
  * </p>
  *
  * @version 1.0.0
  *
  */

class Demo01 extends Actor with ActorLogging {


  override def preStart(): Unit = log.info("IoT Application started")

  override def postStop(): Unit = log.info("IoT Application stopped")

  override def receive: Receive = {
    Actor.emptyBehavior
  }
}

object Demo01 {

  def props(): Props = Props(new Demo01)

  def main(args: Array[String]): Unit = {
    val system = ActorSystem("iot-system")

    try {
      // Create top level supervisor
      val supervisor = system.actorOf(Demo01.props(), "iot-supervisor")
      // Exit the system after ENTER is pressed
      StdIn.readLine()
    } finally {
      system.terminate()
    }
  }
}

File: akka/src/test/java/com/ws/akka/test/demo03/Demo01.java
package com.ws.akka.test.demo03;

import akka.actor.ActorRef;
import akka.actor.ActorSystem;
import org.junit.Test;

import java.util.concurrent.Executor;

/**
 * <p>
 * 类名称：Server
 * </p>
 * <p>
 * 类描述：${DESCRIPTION}
 * </p>
 * <p>
 * 创建人：sun
 * </p>
 * <p>
 * 创建时间：2018-09-03 9:23
 * </p>
 * <p>
 * 修改人：
 * </p>
 * <p>
 * 修改时间：
 * </p>
 * <p>
 * 修改备注：
 * </p>
 * <p>
 * Copyright (c) 版权所有
 * </p>
 *
 * @version 1.0.0
 */
public class Demo01 {

    @Test
    public void test01() {
        ActorSystem system = ActorSystem.create("demo01");
        ActorRef actorOf = system.actorOf(com.ws.akka.demo03.Demo01.create(), "actorRefByDemo01");


    }
}
